automation/all_clippy_checks.sh

```diff
@@ -17,7 +17,10 @@ fi
 
 EXTRA_ARGS=( "$@" )
 
-cargo clippy --all --all-targets --all-features -- -D warnings ${EXTRA_ARGS[@]:+"${EXTRA_ARGS[@]}"}
+# Later rust versions downgraded some warning to pedantic, so we allow them here.
+ALLOWS=("-Aclippy::unreadable-literal"  "-Aclippy::trivially-copy-pass-by-ref" "-Aclippy::match-bool")
```

Comment:
Should this list be adjusted over time as we update the version of rust, and is there any way we can leave reminders to ourselves to do this?

Reply:
Good question but I can't think of a way. I guess that ideally they'd be next to the single place where we specify the rust version, but I can't see a way to avoid duplicating that either.

However, the "good" thing here is that this is listing things which later version of rust are *less* strict about - so there's no harm in keeping them there. This is better than if they were things which were made more strict in later versions.